source/sessions/driver-sessions.rst

```diff
@@ -501,10 +504,17 @@ include a session ID in a ``KILLCURSORS`` command.
 
 Sessions and Connections
 ========================
-A driver MUST only obtain an implicit session after it successfully checks out a connection.
-This limits the number of implicit sessions to never exceed the maximum connection pool size.
-The motivation for this behavior is to prevent too many sessions from being created in a scenario
-where only a limited number are actually needed to execute operations (i.e. TooManyLogicalSessions error).
+A driver MUST only obtain an implicit session's ``ServerSession`` after it successfully checks out a connection.
+A driver SHOULD NOT attempt to release the acquired session before connection check in.
+There are a variety of cases, such as, retryable operations or cursor creating operations
+where a ``serverSession`` must remain acquired by the ``ClientSession`` after an operation is attempted.
+Attempting to account for all these scenarios has risks that do not justify the potential guaranteed ``ServerSession`` allocation limiting.
+
+Drivers SHOULD attempt to release the ``ServerSession`` to the pool at the earliest possible opportunity.
+Drivers SHOULD attempt to reuse ``ServerSession`` as best as possible, concurrency fairness willing.
+
+Through both through the pooling mechanism and limiting acquisition to successful
+connection checkout we can have guaranteed improvement of ``ServerSession`` recycling.
 
 Explicit sessions MAY be changed to allocate a server session similarly, but it is not required.
```

Comment:
The "but it is not required" part is redundant since "MAY" already means not required.

Reply:
It's just really, really, really optional 🙂  (fixed)